src/server/scripts/World/npc_innkeeper.cpp

```diff
@@ -15,155 +15,308 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-/* ScriptData
-SDName: Npc_Innkeeper
-SDAuthor: WarHead
-SD%Complete: 99%
-SDComment: Complete
-SDCategory: NPCs
-EndScriptData */
-
+#include "GameEventMgr.h"
+#include "Player.h"
 #include "ScriptMgr.h"
 #include "ScriptedCreature.h"
 #include "ScriptedGossip.h"
-#include "GameEventMgr.h"
-#include "Player.h"
 #include "WorldSession.h"
 
-enum Spells
+enum InnkeeperSpells
 {
-    SPELL_TRICK_OR_TREATED      = 24755,
-    SPELL_TREAT                 = 24715
+    SPELL_TRICK_OR_TREATED                          = 24755,
+    SPELL_TREAT                                     = 24715,
+
+    SPELL_TRICK                                     = 24753,
+    SPELL_LEPPER_GNOME_COSTUME                      = 24713,
+    SPELL_MALE_GHOST_COSTUME                        = 24735,
+    SPELL_FEMALE_GHOST_COSTUME                      = 24736,
+    SPELL_MALE_NINJA_COSTUME                        = 24710,
+    SPELL_FEMALE_NINJA_COSTUME                      = 24711,
+    SPELL_MALE_PIRATE_COSTUME                       = 24708,
+    SPELL_FEMALE_PIRATE_COSTUME                     = 24709,
+    SPELL_SKELETON_COSTUME                          = 24723,
+    SPELL_HALLOW_END_CANDY_1                        = 24924,
+    SPELL_HALLOW_END_CANDY_2                        = 24925,
+    SPELL_HALLOW_END_CANDY_3                        = 24926,
+    SPELL_HALLOW_END_CANDY_4                        = 24927
 };
 
 #define LOCALE_TRICK_OR_TREAT_0 "Trick or Treat!"
+#define LOCALE_TRICK_OR_TREAT_1 "사탕 하나 주면 안 잡아먹지!"
 #define LOCALE_TRICK_OR_TREAT_2 "Des bonbons ou des blagues!"
 #define LOCALE_TRICK_OR_TREAT_3 "Süßes oder Saures!"
+#define LOCALE_TRICK_OR_TREAT_4 "没有糖果就捣乱！"
+#define LOCALE_TRICK_OR_TREAT_5 "不給糖就搗蛋!"
```

Comment:
I think the best thing would be to remove the local ones from here and take the ones that exist within the DB in `broadcast_text` in such a way as to reduce the amount of code used.

```
SELECT * FROM `broadcast_text` WHERE `id` = '10693';
```

Reply:
I closed the PR because, even though the initial reasoning was updating the whole script to Trinity's new registration model and standards, there's already support for each innkeeper (and also for Hallow's End) in the DB and this code seems deprecated to me. It can be used for innkeepers whose conditions and gossip_menu_options are not set on the DB as a "life-saver" until they get their stuff implemented.

Today, I did some research and each individual innkeeper needs to get their conditions and gossip_options looked at since each of them has a different OptionIndex for each option (setting your inn, browsing goods, trick or treat, etc.), and many of these were sniffed from WoTLK and there were changes in Cataclysm. In fact, the gossip_option selection is handled, mostly, by SmartAI, but it's wrongly set most of the cases because of this (e.g. you click on browse your goods and you receive a hallow's end debuff). 

If you take a look at 0a4ccbe4275b6c0dc2a10c96b7d8529ffb1d9bb5, I implemented support for several innkeepers who didn't have gossip_menu_options or had them wrong.

Reply:
It simply gave the point of view in which you could save code and not have to load all the Locales in the script and at the time of calling the gossip take the data directly from the DB